src/Chunk.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+*  @file    Chunk.cpp
+*  @author  
+*  @date    05 June 2018
+*/
+
+#include "Chunk.h"
+#include "Perlin.h"
+
+#define Z_VALUE 0
+
+Chunk::Chunk(u8 x, u8 y, Area areas[], u8 type) : x(x * CHUNK_SIZE), y(y * CHUNK_SIZE), type(type)
+{
+	// Ich bin irgendwie zu dumm das richtig zu machen XD
+    this->areas[0] = areas[0];
+    this->areas[1] = areas[1];
+    this->areas[2] = areas[2];
+    this->areas[3] = areas[3];
+
+	if (type < 0 || type > 3)
+		throw std::invalid_argument("Illegal chunk type - must be in between 0 and 3");
+}
+
+void Chunk::calculate() // Hier k�nnte man evtl. Z �bergeben, falls man sp�ter doch wieder animieren will..
+{
+	switch (type)
+	{
+		case 0: calculate_inner(); break;
+		case 1: calculate_vertical(); break;
+		case 2: calculate_horizontal(); break;
+		case 3: calculate_corner(); break;
+		default: throw std::invalid_argument("Illegal chunk type - must be in between 0 and 3"); break;
+	}
+}
+
+void Chunk::calculate_inner()
+{
+    for (int xx = 0; xx < CHUNK_SIZE; ++xx)
+    {
+        for (int yy = 0; yy < CHUNK_SIZE; ++yy)
+        {
+            values[xx * CHUNK_SIZE + yy] = octavePerlin(xx + x, yy + y, Z_VALUE, areas[0]);
+        }
+    }
+}
+
+void Chunk::calculate_vertical()
+{
+    Area upper_area = areas[0];
+    Area lower_area = areas[1];
+
+    for (int xx = 0; xx < CHUNK_SIZE; ++xx)
+    {
+        for (int yy = 0; yy < CHUNK_SIZE; ++yy)
+        {
+            float upper = octavePerlin(xx + x, yy + y, Z_VALUE, upper_area);
+            float lower = octavePerlin(xx + x, yy + y, Z_VALUE, lower_area);
+
+            values[xx * CHUNK_SIZE + yy] = 0; // interpolieren ziwschen upper & lower in Abh�ngigkeit von yy
```

Comment:
Interpolation in Abhängigkeit von Y fehlt natürlich.

Wenn ichs mir grad so anschauen, kann es gut sein, dass ich hier von der Logik her X und Y vertauscht habe....... XD